### ISSUE TITLE
How to extract or plot gradient for a specific layer in tensorpack?

### ISSUE THREAD

**Issue by tongwu92623** [tensorpack/tensorpack#935]
Hi,

How could I extract the gradient of each layer during training in tensorpack?

Best regards,

Tong Wu

**Comment by ppwwyyxx** [comment]
`tf.gradients(cost, some_tensor)`

**Comment by ppwwyyxx** [comment]
You can also wrap a `tf.identity` function with `tf.custom_gradient`. This will let you access to the gradients in the backward pass and is more efficient in runtime. You can refer to tensorflow documentation to use it or `tf.gradients`.

**Comment by ppwwyyxx** [comment]
Closing as there is no response, and how to get gradients w.r.t. each layer's output is a tensorflow's question unrelated to tensorpack.
